presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java

```diff
@@ -54,6 +54,8 @@ protected boolean supportsNotNullColumns()
     public void close()
             throws Exception
     {
+        // Await query events to propagate and finish
+        Thread.sleep(1000);
```

Comment:
We should put this after `runningQueryIds`.
```java
...
Set<QueryId> runningQueryIds = ((TestingHiveEventListener) eventListener.get()).getRunningQueries();
if (!runningQueryIds.isEmpty()) {
    // sleep
}

// it's better to print more info in case of failures.
assertEquals(
    runningQueryIds,
    ImmutableSet.of(),
    format(
            "Query completion events not sent for %d queries: %s",
            runningQueryIds.size(),
            runningQueryIds.stream().map(QueryId::getId).collect(joining(", "))));